napari_animation/animation.py

```diff
@@ -1,334 +1,171 @@
 import imageio
-import copy
 import skimage.transform
 import skimage.io
 import numpy as np
+from copy import deepcopy
 from pathlib import Path
 
-from . import utils
+from .utils import interpolate_state
 
 
 class Animation:
-    def __init__(self, viewer=None, inter_steps=15):
+    """Make animations using the napari viewer.
+
+    Parameters
+    ----------
+    viewer : napari.Viewer
+        napari viewer.
+
+    Attributes
+    ----------
+    key_frames : list of dict
+        List of viewer state dictionaries.
+    frame : int
+        Currently shown key frame.
+    """
+    def __init__(self, viewer):
+        self.viewer = viewer
 
-        """Standard __init__ method.
+        self.key_frames = []
+        self.frame = -1
 
+    def capture_keyframe(self, steps=30, ease=None, insert=True, frame=None):
+        """Record current key-frame
+        
         Parameters
         ----------
-        viewer : viewer
-            napari viewer
-        inter_steps: int
-            number of steps to interpolate between key frames
-
-        Attributes
-        ----------
-        key_frames : list
-            list of dictionary defining viewer states. Dictionaries have keys:
-                'frame': int, frame
-                'ndisplay': int, view type (2,3)
-                'camera': camera state
-                'vis': list of booleans, visibility of layers
-                'sliders': list, slider positions
-
-        interpolated_states: dict
-            dictionary defining interpolated states. Each element is a list of length N
-            frames. Keys are:
-                'ndisplay': int, view type (2,3)
-                'camera': camera state
-                'vis': list of booleans, visibility of layers
-                'sliders': list, slider positions
-
-        states_dict : list
-            list of dictionaries defining viewer states for each frame. Same keys as key_frames
-
-        current_frame : int
-            currently shown key frame
-        implot : matplotlib Ax object
-            reference to matplotlib image used for movie returned by imshow
-        anim : matplotlib FuncAnimation object
-            reference to animation object
+        steps : int
+            Number of interpolation steps between last keyframe and captured one.
+        ease : callable, optional
+            If provided this method should make from `[0, 1]` to `[0, 1]` and will
+            be used as an easing function for the transition between the last state
+            and captured one.
```

Comment:
Here's a nice [gist of easing functions in python](https://gist.github.com/zeffii/c1e14dd6620ad855d81ec2e89a859719). We could expose a subset of these as an ENUM in a dropdown menu for GUI users